common/static/coffee/spec/discussion/view/discussion_thread_view_spec.coffee

```diff
@@ -12,13 +12,40 @@ describe "DiscussionThreadView", ->
         spyOn(DiscussionThreadShowView.prototype, "convertMath")
         spyOn(DiscussionContentView.prototype, "makeWmdEditor")
         spyOn(DiscussionUtil, "makeWmdEditor")
-        spyOn(ThreadResponseView.prototype, "renderShowView")
+        spyOn(DiscussionUtil, "setWmdContent")
+        spyOn(ThreadResponseShowView.prototype, "convertMath")
 
     renderWithContent = (view, content) ->
-        DiscussionViewSpecHelper.setNextResponseContent(content)
+        $.ajax.andCallFake((params) =>
+            params.success(
+                createAjaxResponseJson(content, false),
+                'success'
+            )
+            {always: ->}
+        )
         view.render()
         jasmine.Clock.tick(100)
 
+    renderWithTestResponses = (view, count, options) ->
+        renderWithContent(
+            view,
+            _.extend(
+                {
+                    resp_total: count,
+                    children: if count > 0 then (createTestResponseJson(index) for index in [1..count]) else []
+                },
+                options
+            )
+        )
+
+    createTestResponseJson = (index) ->
+        {
+            user_id: window.user.id,
+            body: "Response " + index,
+            id: "id_" + index,
+            created_at: "2015-01-01T22:20:28Z"
+        }
```

Comment:
Big thanks to @andy-armstrong for writing these two test functions, we wouldn't have been able to reproduce the buggy behavior without these being added. There are several other places in this PR that have been updated to use this new system, instead of constructing "responses" as dicts like this: `renderWithContent(view, {resp_total: 1, children: [{}]})`